### PR TITLE
DataTable

### PR DESCRIPTION
**Pull request by jordankoschei-okta**

### CODE REVIEW

File: packages/odyssey-react-mui/src/labs/DataTable.tsx [hunk 1]
```diff
@@ -0,0 +1,646 @@
...
+      showSkeletons,
+    },
+    rowVirtualizerInstanceRef: rowVirtualizerInstanceRef,
+    rowVirtualizerOptions: { overscan: 4 },
+    enableRowVirtualization:
+      paginationType === "loadMore" || resultsPerPage > 50,
+    enableColumnResizing: hasColumnResizing,
+    enableDensityToggle: false,
+    enableFullScreenToggle: false,
+    enablePagination: false,
```
Comment (benschell-okta) [inline packages/odyssey-react-mui/src/labs/DataTable.tsx:280]: Do you also need `manualPagination: false` (just like `enableFilters`/`manualFiltering`)?

**Comment by benschell-okta** [comment]
Not a code thing: I cloned the branch and ran the storybook locally. I'm really digging the custom impls of sorting and pagination, especially having tried the existing impl. 

With that said: is there a way to improve the column filtering when the contents of the cell is markup? I tried filtering by "Risk level" and typed "Low" or "Medium" in the box but it got no results.

**Comment by benschell-okta** [comment]
One other possible bug: in storybook, on the DataFilters Docs page, the last filter option in the dropdown for "Autocomplete filter" just renders an empty box:
Screenshot 2023-11-14 at 9 56 23 AM

File: packages/odyssey-react-mui/src/Button.tsx [hunk 2]
```diff
@@ -164,13 +164,13 @@ const Button = ({
...
     <>
-      {tooltipText && (
+      {tooltipText && !isDisabled && (
         <Tooltip ariaType="description" placement="top" text={tooltipText}>
           <MuiPropsContext.Consumer>{renderButton}</MuiPropsContext.Consumer>
         </Tooltip>
       )}
 
-      {!tooltipText && renderButton(muiProps)}
+      {!(tooltipText && !isDisabled) && renderButton(muiProps)}
```
Comment (benschell-okta) [inline packages/odyssey-react-mui/src/Button.tsx:173]: I'm not a regular contributor here, but I'd recommend this for readability (rather than having to mentally do the double-negative):

`{ (!tooltipText || isDisabled) && renderButton(muiProps) }`

**Review by benschell-okta — APPROVED** [review]
I'm very excited about this. Out of personal curiosity: what is the process or requirements for something like DataTable moving out of Labs?

**Comment by jordankoschei-okta** [comment]
Hey @benschell-okta, I'm pumped you're trying this!

Some of the filtering/search/sorting weirdness isn't part of `<DataTable>` itself — the approach I'm pursuing here is "bring-your-own-logic," so any data operations are provided by the consumer. In this case, in Storybook, _I'm_ the consumer, and some of the dummy code I wrote for filtering and sorting is... uh, admittedly not great. If you find any jankiness there, there's a decent chance it's my fault but not part of `<DataTable>` 😬 

We're hoping to move this out of Labs pretty quickly. The remaining things we'll need to do are:
1. Fix some broken RTL support
2. Add proper keyboard/a11y support to the drag-and-drop reordering
3. Replace some magic values with their corresponding tokens (and add tokens, if applicable)
4. Replace hard-coded English-language strings with translation tokens
5. Refactor some ugly code that could be more efficient (we've got some missing `useCallback`s, etc)

We also want a handful of designers and engineers to poke around and see if they can break it... another reason I'm glad you're already taking a look around!

File: packages/odyssey-react-mui/src/Button.tsx (hunk 2 above)
Comment (jordankoschei-okta) [reply packages/odyssey-react-mui/src/Button.tsx:173]: That's a big improvement, I'll make that change! 🙌

Comment (jordankoschei-okta) [reply packages/odyssey-react-mui/src/Button.tsx:173]: And we welcome all suggestions, no matter how regular or irregular a contributor you are. The more, the better! 🚀

**Comment by benschell-okta** [comment]
Awesome, thanks @jordankoschei-okta 

So far I've only tried this locally on storybook. I do have a real impl in `atko-eng/ui-devtools` repo and will have some less busy time next week, so I'll give this a try and report back with any other findings.

File: packages/odyssey-react-mui/package.json [hunk 3]
```diff
@@ -73,7 +73,7 @@
     "@okta/odyssey-design-tokens": "workspace:*",
     "date-fns": "^2.30.0",
     "i18next": "^23.5.1",
-    "material-react-table": "^1.14.0",
+    "material-react-table": "^2.0.2",
```
[the remark this answers is not in the document]
Comment (KevinGhadyani-Okta) [reply packages/odyssey-react-mui/package.json:76]: Wow, I was afraid to do this upgrade after the library author DM'd me about it :P. Nice to see it here!

File: packages/odyssey-react-mui/src/Button.tsx (hunk 2 above)
Comment (KevinGhadyani-Okta) [reply packages/odyssey-react-mui/src/Button.tsx:173]: I would even swap it more:
```ts
{ (isDisabled || !tooltipText) && renderButton(muiProps) }
```

Put the positive check first.

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 4]
```diff
@@ -49,10 +64,12 @@ export type MenuButtonProps = {
...
-  children: Array<
-    | ReactElement<typeof MenuItem | typeof Divider | typeof ListSubheader>
-    | NullElement
-  >;
+  children:
+    | ReactFragment
+    | Array<
+        | ReactElement<typeof MenuItem | typeof Divider | typeof ListSubheader>
+        | NullElement
+      >;
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:72]: Do we want `NullElement` as a possibility outside of the array as well?

Comment (KevinGhadyani-Okta) [reply packages/odyssey-react-mui/src/MenuButton.tsx:72]: Also, make sure you've merged in `main` because I think this part has changed.

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 5]
```diff
@@ -65,10 +82,25 @@ export type MenuButtonProps = {
...
+   */
+  menuAlignment?: "left" | "right";
+  /**
+   * Optional function to fire on button click
+   */
+  onClick?: ButtonProps["onClick"];
+  /**
+   * Optional function to fire on memu close
+   */
+  menuPaperRef?: MutableRefObject<HTMLDivElement>;
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:96]: Why do we need this ref?

Comment (KevinGhadyani-Okta) [reply packages/odyssey-react-mui/src/MenuButton.tsx:96]: If we need a ref, maybe we can make an internal version of this component and expose only the one without this ref?

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 6]
```diff
@@ -65,10 +82,25 @@ export type MenuButtonProps = {
...
+  onClose?: MenuProps["onClose"];
+  preventCloseOnChildClick?: boolean;
   /**
    * The size of the button
    */
   size?: (typeof buttonSizeValues)[number];
+  forceMenuState?: boolean;
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:103]: We should use an `is` or `has` prefix. So this would be what? I can't tell what it's doing. We should also add a comment.

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 7]
```diff
@@ -65,10 +82,25 @@ export type MenuButtonProps = {
...
+  /**
+   * Optional function to fire on button click
+   */
+  onClick?: ButtonProps["onClick"];
+  /**
+   * Optional function to fire on memu close
+   */
+  menuPaperRef?: MutableRefObject<HTMLDivElement>;
+  onClose?: MenuProps["onClose"];
+  preventCloseOnChildClick?: boolean;
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:98]: This should also be `is` or `has`. Something like `hasCloseButton` which currently defaults to `true`. Or `isClosable`; although, that's less-effective.

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 8]
```diff
@@ -65,10 +82,25 @@ export type MenuButtonProps = {
    * If the MenuButton is an overflow menu or standard menu.
    */
   isOverflow?: boolean;
+  /**
+   * The horizontal alignment of the menu.
+   */
+  menuAlignment?: "left" | "right";
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx]: Don't we want this to be an array of values that we set as a type and pull into Storybook?

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 9]
```diff
@@ -110,13 +148,25 @@ const MenuButton = ({
...
-    setAnchorEl(null);
-  }, []);
+  const closeMenu = useCallback(
+    (event?, reason?) => {
+      if (preventCloseOnChildClick && reason !== "backdropClick") {
+        return;
+      }
+
+      onClose?.(event, reason);
+      setAnchorEl(null);
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx]: `setAnchorElement` would be more clear.

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 10]
```diff
@@ -110,13 +148,25 @@ const MenuButton = ({
...
+    },
+    [preventCloseOnChildClick, onClose]
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx]: Why are `event and `reason` marked as optional args? Aren't they always passed?

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 11]
```diff
@@ -110,13 +148,25 @@ const MenuButton = ({
...
   const isOpen = Boolean(anchorEl);
 
-  const closeMenu = useCallback<MenuContextType["closeMenu"]>(() => {
-    setAnchorEl(null);
-  }, []);
+  const closeMenu = useCallback(
+    (event?, reason?) => {
+      if (preventCloseOnChildClick && reason !== "backdropClick") {
+        return;
+      }
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx]: Why is this an optional thing? What's the reasoning behind sometimes closing or not closing the menu based on if a child was clicked? Is this so you can open sub-menus?

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 12]
```diff
@@ -110,13 +148,25 @@ const MenuButton = ({
...
-  const openMenu = useCallback<MenuContextType["openMenu"]>((event) => {
-    setAnchorEl(event.currentTarget);
-  }, []);
+  const openMenu = useCallback(
+    (event) => {
+      onClick?.(event);
+      setAnchorEl(event.currentTarget);
+    },
+    [onClick]
+  );
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx]: Why did we remove the generic passed in for this function?

File: packages/odyssey-react-mui/src/MenuButton.tsx (hunk 10 above)
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx]: We should type the `useCallback` with a generic so our intent is clear.

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 13]
```diff
@@ -162,10 +219,15 @@ const MenuButton = ({
 
       <Menu
         anchorEl={anchorEl}
+        anchorOrigin={{ horizontal: menuAlignment, vertical: "bottom" }}
+        transformOrigin={{ horizontal: menuAlignment, vertical: "top" }}
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:182]: Can we put these in `useMemo` so they don't cause unnecessary re-renders?

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 14]
```diff
@@ -162,10 +219,15 @@ const MenuButton = ({
...
         id={`${uniqueId}-menu`}
         MenuListProps={menuListProps}
         onClose={closeMenu}
         open={isOpen}
+        PaperProps={{
+          ref: menuPaperRef,
+        }}
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx]: Yeah, this is why

File: packages/odyssey-react-mui/src/MenuButton.tsx [hunk 15]
```diff
@@ -162,10 +219,15 @@ const MenuButton = ({
 
       <Menu
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/MenuButton.tsx:180]: Is this MUI's `Menu` component? I know it's not a big deal, but could we rename it to `MuiMenu`? It's confusing who's menu is who's when skimming code.

File: packages/odyssey-react-mui/src/SearchField.tsx [hunk 16]
```diff
@@ -37,6 +37,15 @@ export type SearchFieldProps = {
    * If `true`, the component will receive focus automatically.
    */
   hasInitialFocus?: boolean;
+  /**
+   * The id of the `input` element.
+   */
+  id?: string;
+  inputRef?: Ref<HTMLInputElement>;
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/SearchField.tsx:44]: Why are we adding an `inputRef` here?

Comment (KevinGhadyani-Okta) [reply packages/odyssey-react-mui/src/SearchField.tsx:44]: Can we pass stuff into `SearchField` that makes it work with the table rather than passing in the whole ref?

Even if we need a ref, we should use `useImperativeHandle` instead of passing the ref directly down.

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 17]
```diff
@@ -0,0 +1,540 @@
...
+  "autocomplete",
+  "checkbox",
+  "date",
+  "date-range",
+  "multi-select",
+  "range",
+  "range-slider",
+  "select",
+  "text",
+] as const;
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/DataFilters.tsx:52]: Not saying we change it, but based on the grammar here, shouldn't this be `"auto-complete"`? If so, I'm fine changing it. No reason to match the component name, but the component name implies this is a word.
```suggestion
export const dataFilterVariantValues = [
  "auto-complete",
  "checkbox",
  "date",
  "date-range",
  "multi-select",
  "range",
  "range-slider",
  "select",
  "text",
] as const;
```

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 18]
```diff
@@ -0,0 +1,540 @@
...
+
+export type DataFiltersProps = {
+  onChangeSearch?: (value: string) => void;
+  onChangeFilters?: (filters: Array<DataFilter>) => void;
+  searchOnSubmit?: boolean;
+  searchDelay?: number;
+  initialSearchTerm?: string;
+  additionalActions?: ReactNode;
+  filters?: Array<DataFilter>;
+};
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/DataFilters.tsx:70]: Can you add some documentation to these types? I'm seeing `value` under `DataFilter`, and I don't know what it does.

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 19]
```diff
@@ -0,0 +1,540 @@
...
+  label: string;
+  variant?: (typeof dataFilterVariantValues)[number];
+  value?: string | string[] | undefined;
+  options?: Array<{ label: string; value: string }>;
+};
+
+export type DataFiltersProps = {
+  onChangeSearch?: (value: string) => void;
+  onChangeFilters?: (filters: Array<DataFilter>) => void;
+  searchOnSubmit?: boolean;
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/DataFilters.tsx:65]: This needs to have an `is` or `has` prefix. I don't know what it's doing, so I can't come up with anything off the top of my head.

Typically, when you're saying "search on submit", that's always `true`. It's the name of a function; a command. Which makes it confusing because we're not calling a function, we're defining a state.

Using `is` or `has` help define the state or features of a component. Either they have the feature or they don't. We're not telling the component to perform an action, we're flicking a light switch.

If you called `DataFilters.searchOnSubmit()`, then it would make more sense with that grammar, but technically, listeners should start with `on` to make it clear we're performing an action that could occur over time :p.

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 20]
```diff
@@ -0,0 +1,540 @@
...
+  additionalActions?: ReactNode;
+  filters?: Array<DataFilter>;
+};
+
+const DataFilters = ({
+  onChangeSearch,
+  onChangeFilters,
+  searchOnSubmit = false,
+  searchDelay = 200,
+  initialSearchTerm,
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/DataFilters.tsx:77]: Should this be `initialSearchTerm` or `defaultSearchTerm` like we have in other areas of the code?

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 21]
```diff
@@ -0,0 +1,540 @@
...
+      (
+        accumulator: { [key: string]: string | string[] | undefined },
+        filter: DataFilter
+      ) => {
+        accumulator[filter.id] = filter.value;
+        return accumulator;
+      },
+      {}
+    )
+  );
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/DataFilters.tsx:95]: We should `useMemo` this initial value. It's a micro-optimization, but I don't like the idea of calling `reduce` before every re-render for a performance-sensitive item like a table.

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 22]
```diff
@@ -0,0 +1,540 @@
...
+  }>(
+    filtersProp.reduce(
+      (
+        accumulator: { [key: string]: string | string[] | undefined },
+        filter: DataFilter
+      ) => {
+        accumulator[filter.id] = filter.value;
+        return accumulator;
+      },
+      {}
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/DataFilters.tsx:93]: We shouldn't be defining the types for `accumulator` nor `filter`. The type for `filter` should be coming from `filtersProp`.

The fix for `accumulator`'s type is using `{} as` in the second arg (initial state):
```suggestion
    filtersProp.reduce(
      (
        accumulator: { [key: string]: string | string[] | undefined },
        filter: DataFilter
      ) => {
        accumulator[filter.id] = filter.value;
        return accumulator;
      },
      {} as Record<string, string | string[] | undefined>,
```

But also, why is it `string | string[] | undefined`? I don't even know what these filters are regardless of this being typed. When looking at this part of the code, I should know what I'm dealing with and what a `string | string[] | undefined` means.

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 23]
```diff
@@ -0,0 +1,540 @@
...
+    )
+  );
+  const [search, setSearch] = useState<string>(initialSearchTerm ?? "");
+  const activeFilters = filters.filter((filter) => filter.value);
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/DataFilters.tsx:97]: Does this properly filter down types or do we need to setup a predicate? No, that's not the word. There's some special term where you use the TypeScript keyword `is` in a filter function.

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 24]
```diff
@@ -0,0 +1,540 @@
...
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
+  const [menuAnchorEl, setMenuAnchorEl] = useState<HTMLElement | undefined>(
+    undefined
+  );
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/DataFilters.tsx:101]:
```suggestion
  const [menuAnchorElement, setMenuAnchorElement] = useState<HTMLElement | undefined>(
    undefined
  );
```

_Which_ `menuAnchorElement` is this? Why can it sometimes be `undefined`? I can't tell from reading this part of the code. Is it the selected menu anchor element? Is it the only one? Based on the way it's written, I would expect there to only be one, but then why is there a state?

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 25]
```diff
@@ -0,0 +1,540 @@
...
+  const [popoverState, setPopoverState] = useState<{
+    isOpen: boolean;
+    anchorEl?: HTMLElement;
+    filter?: DataFilter;
+  }>({
+    isOpen: false,
+  });
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/DataFilters.tsx:108]: I think it might be more clear here to define the other properties as `undefined`:
```suggestion
  const [popoverState, setPopoverState] = useState<{
    isOpen: boolean;
    anchorEl?: HTMLElement;
    filter?: DataFilter;
  }>({
    anchorElement: undefined,
    filter: undefined,
    isOpen: false,
  });
```

But really, I don't understand why these are here in the first place. This looks like 3 states, not 1, and I'd prefer to see them as a separate states in that case. But it might be 1 state too. I don't have enough context to say either way.

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 26]
```diff
@@ -0,0 +1,540 @@
...
+  initialSearchTerm?: string;
+  additionalActions?: ReactNode;
+  filters?: Array<DataFilter>;
+};
+
+const DataFilters = ({
+  onChangeSearch,
+  onChangeFilters,
+  searchOnSubmit = false,
+  searchDelay = 200,
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/labs/DataFilters.tsx:76]: Maybe this should be named `searchDelayTime`. This one's a minor nit-pick.

If it was me, I'd name it `searchDelayTimeInMilliseconds`, but that's typically not a name I'd use on a component prop because you can document "it's in milliseconds" on the prop. If it was an internal variable, it's best to have `InMilliseconds` because that's _not_ documented anywhere.

**Review by KevinGhadyani-Okta — CHANGES_REQUESTED** [review]
I'm gonna stop here and say we should probably have a meeting instead.

File: packages/odyssey-react-mui/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (jordankoschei-okta) [reply packages/odyssey-react-mui/package.json:76]: It let us remove so many workarounds!

File: packages/odyssey-react-mui/src/labs/DataTable.tsx (hunk 1 above)
Comment (jordankoschei-okta) [reply packages/odyssey-react-mui/src/labs/DataTable.tsx:281]: We don't, quite unintuitively, because of the way we're loading in the data — the pagination's being handled by our fetch function.

**Review by KevinGhadyani-Okta — COMMENTED** [review]
I'm going on vacay, so leaving this to other devs.

**Review by KevinGhadyani-Okta — APPROVED** [review]
I have to approve this to remove "requested changes". Please communicate with Chris first.

File: packages/odyssey-react-mui/src/labs/DataTable.tsx [hunk 27]
```diff
@@ -0,0 +1,687 @@
...
+                    />
+                    {column.header}
+                  </MenuItem>
+                ))}
+            </>
+          </MenuButton>
+        )}
+      </>
+    ),
+    [density, columnVisibility, columns]
```
Comment (chrispulsinelli-okta) [inline packages/odyssey-react-mui/src/labs/DataTable.tsx:635]: You're missing some dependencies in this callback.

File: packages/odyssey-storybook/package.json [hunk 28]
```diff
@@ -13,14 +13,17 @@
     "url": "https://github.com/okta/odyssey",
     "directory": "packages/odyssey-storybook"
   },
+  "dependencies": {
+    "@faker-js/faker": "^8.2.0",
+    "@okta/odyssey-react-mui": "workspace:*"
```
Comment (chrispulsinelli-okta) [inline packages/odyssey-storybook/package.json:18]: Why is this moved up to `dependencies`?
Should `faker` also be a devDependency?

**Review by chrispulsinelli-okta — COMMENTED** [review]
_tl;dr not blocking this as I believe you already have some linting issues preventing a merge that need to be fixed, but I have general concerns about the readability and declaration of arrow functions in components in this PR._

You have some callbacks to fix with either missing or unnecessary props being provided. I also left some general concerns and recommendations that you may want to follow up with or fix now as well.

I have time today to chat about this PR with you and would be happy to go over my comments with you if you have time.

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 29]
```diff
@@ -0,0 +1,577 @@
...
+   * and `onChangeSearch` will fire when the button is clicked, rather than
+   * whenever the input value changes.
+   */
+  hasSearchSubmitButton?: boolean;
+  /**
+   * The debounce time, in milliseconds, for the search input firing
+   * `onChangeSearch` when changed. If `hasSearchSubmitButton` is true,
+   * this doesn't do anything.
+   */
+  searchDelayTime?: number;
```
[the remark this answers is not in the document]
Comment (jordankoschei-okta) [reply packages/odyssey-react-mui/src/labs/DataFilters.tsx:102]: The default value (200ms, IIRC) is a nice middle-ground that feels responsive without making excessive API calls, but in some use cases I could see it going closer to 500ms or 100ms.

File: packages/odyssey-react-mui/src/labs/DataFilters.tsx [hunk 30]
```diff
@@ -0,0 +1,577 @@
...
+  searchDelayTime = 200,
+  defaultSearchTerm = "",
+  additionalActions,
+  filters: filtersProp = [],
+}: DataFiltersProps) => {
+  const [filters, setFilters] = useState<DataFilter[]>(filtersProp);
+
+  const initialInputValues = useMemo(() => {
+    return filtersProp.reduce((accumulator, filter) => {
+      accumulator[filter.id] = filter.value;
```
[the remark this answers is not in the document]
Comment (jordankoschei-okta) [reply packages/odyssey-react-mui/src/labs/DataFilters.tsx:133]: It shouldn't be possible to have two filters with the same id; if there are, it means there's an underlying problem with the consumer's data structure.
